### PR TITLE
Setup new Development Workspace + Automated Deploy

### PR DESCRIPTION
**Pull request by brunoocasali**
Fixes the issue #338

### CODE REVIEW

File: app/controllers/allies_controller.rb [hunk 1]
```diff
@@ -109,15 +109,4 @@ def remove
       format.json { head :no_content }
     end
   end
-
-  private
-
-  def if_not_signed_in
```
Comment (julianguyen) [inline app/controllers/allies_controller.rb]: Why is this removed?

File: config/application.yml [hunk 2]
```diff
@@ -8,13 +8,13 @@
 # makes 'Your_Gmail_Username' available as ENV["GMAIL_USERNAME"]
 
 #This is the only configuration value required to boot the application
-SECRET_KEY_BASE: "examplesecretkey123123"
+SECRET_KEY_BASE: "945a3eb9dfc2b5ffa394d3e084dc49e08be6a548fc926976bbc9d8a61adc917214e0c85b20288550cfaf0384dcdde3b07e91f851b1b91a7739f4d67a58b882ae"
```
Comment (julianguyen) [inline config/application.yml]: Think we should revert this line

File: config/application.yml [hunk 3]
```diff
@@ -8,13 +8,13 @@
...
 
 # Configuration values for sending email, set these to true, configure your default "from address",
 # and configure smtp.yml in order to send email
 SEND_EMAIL: "false"
 RAISE_DELIVERY_ERRORS: "false"
-DEFAULT_FROM_ADDRESS: "yourmail@gmail.com"
+DEFAULT_FROM_ADDRESS: "brunoocasali@gmail.com"
```
Comment (julianguyen) [inline config/application.yml]: This should be reverted too

File: app/controllers/allies_controller.rb (hunk 1 above)
Comment (brunoocasali) [inline app/controllers/allies_controller.rb]: Check the [`application_controller.rb:46`](https://github.com/julianguyen/ifme/blob/master/app/controllers/application_controller.rb#L46) there is a duplicated method!

File: config/application.yml (hunk 2 above)
Comment (brunoocasali) [inline config/application.yml]: Not this line, the whole file! You changed the gitignore, and I've don't ran a git remove locally... Because this, this file are here!

File: config/application.yml (hunk 3 above)
Comment (brunoocasali) [inline config/application.yml]: The same here!

File: config/env/development.example.env [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Application
+SECRET_KEY_BASE: ""
```
Comment (brunoocasali) [inline config/env/development.example.env]: Why on development env is SECRET_KEY_BASE? and on test is CONFIG_SECRET_KEY_BASE ?

Comment (brunoocasali) [inline config/env/development.example.env]: And they need to be:

SOME_KEY=value # without quotes!

File: config/env/production.env [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Application
```
Comment (brunoocasali) [inline config/env/production.env]: This file is not necessary, because this keys are already on Heroku!

File: config/env/test.example.env [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+CONFIG_SECRET_KEY_BASE: ""
+
+# Cloudinary
+CLOUDINARY_CLOUD_NAME=""
+CLOUDINARY_API_KEY=""
+CLOUDINARY_API_SECRET=""
+
+# Configuration values for sending email, set these to true, configure your default "from address",
+# and configure smtp.yml in order to send email
+SEND_EMAIL: "false"
```
Comment (brunoocasali) [inline config/env/test.example.env]: If you want to use a SEND_EMAIL key, you need to use like: `SEND_EMAIL=false` and declare ENV['SEND_EMAIL'] on destination! ;D

File: config/env/development.example.env (hunk 4 above)
Comment (julianguyen) [inline config/env/development.example.env]: OMG good catch
Yeah should be SECRET_KEY_BASE

Comment (julianguyen) [inline config/env/development.example.env]: would putting `insert_value` instead of `""` be okay?

File: config/env/test.example.env [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+RAISE_DELIVERY_ERRORS: "false"
```
Comment (brunoocasali) [inline config/env/test.example.env]: Check the comment above!

**Comment by julianguyen** [comment]
@brunoocasali Do we want to remove the wercker stuff?

**Comment by brunoocasali** [comment]
I don't know what stuff are you talking about! :D If you found something go ahead!

File: config/env/development.example.env (hunk 4 above)
Comment (julianguyen) [inline config/env/development.example.env]: Does this need to be updated?

Comment (brunoocasali) [inline config/env/development.example.env]: I've updated :D

**Comment by julianguyen** [comment]
Finally merging this! Thanks @brunoocasali for working on this 💯

**Comment by brunoocasali** [comment]
UHUUUUULLLLLLLL \o/ you rocks @julianguyen :metal:
